File: backend-Nijan/src/main/java/com/ecommerce/backendNijan/service/CategoryService.java
package com.ecommerce.backendNijan.service;

import com.ecommerce.backendNijan.entity.CategoryEntity;
import com.ecommerce.backendNijan.model.CategoryDto;
import com.ecommerce.backendNijan.model.ICategory;
import com.ecommerce.backendNijan.response.PageResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface CategoryService {

    /**
     * Get all category.
     *
     * @return list of entity category.
     */
    List<CategoryDto> getAll();

    /**
     * Get all category.
     *
     * @return list of entity category.
     */
    PageResponse<ICategory> getAllCategoryPage(int page, int size, String sortBy,
                                               String sortDir, String searchValue);

    /**
     * Get all category.
     *
     * @param sortBy sortBy
     * @param searchValue searchValue
     * @param pageable pageable
     * @return list of entity category.
     */
    Page<ICategory> getAllCategoryPage(String sortBy, String searchValue, Pageable pageable);

    /**
     * Get all category.
     *
     * @return list of entity category.
     */
    Optional<CategoryDto> getCategoryByPrimaryKey(Long categoryId);

    /**
     * Get all category.
     *
     * @return list of entity category.
     */
    CategoryEntity insert(CategoryDto dto);

    /**
     * Get all category.
     *
     * @return list of entity category.
     */
    CategoryEntity update(CategoryDto dto);
}

File: vite-nijan/src/component/modules/CategoryService.js
// import { API_CATEGORY_PREFIX } from "../api";
import http from "./http-common";

class CategoryService {

    getAll() {
      return http.get('http://localhost:8888/api/category/list');
    }

    getCategoryById(id) {
      return http.get('http://localhost:8888/api/category?category_id=' + id);
    }
  
    getAllPageable(page) {
      return http.get("http://localhost:8888/api/category/all/page?page_no=" + page.pageNo + "&page_size=" + page.pageSize
      + "&sort_direction=" + page.sortDirection + "&sort_by=" + page.sortBy + "&search_value=" + page.searchValue)
    }

    updateCategory(id, data) {
      return http.put('http://localhost:8888/api/category/update/' + id, data);
    }
    
    createCategory(data) {
      return http.post('http://localhost:8888/api/category/register', data);
    }

    deleteCategory(id) {
      return http.delete('http://localhost:8888/api/category/delete?category_id=' + id);
    }
}
export default new CategoryService();